### PR TITLE
[gdal]: Update documentation

### PR DESCRIPTION
**Pull request by gavindidrichsen**
Documentation updated and individual tests added for 24 binaries...

### CODE REVIEW

File: controls/gdal_exists.rb [hunk 1]
```diff
@@ -16,9 +16,129 @@
...
+  end
+  ogrtindex_full_path = File.join(plan_installation_directory.stdout.strip, "bin/ogrtindex")
+  describe file(ogrtindex_full_path) do
+    it { should exist }
+    it { should be_executable }
+  end
+  testepsg_full_path = File.join(plan_installation_directory.stdout.strip, "bin/testepsg")
+  describe file(testepsg_full_path) do
+    it { should exist }
+    it { should be_executable }
```
Comment (MindNumbing) [inline controls/gdal_exists.rb]: Could you combine these tests with a loop? An example can be viewed [here](https://github.com/chef-base-plans/python2/blob/07207f3ce5f3fe713ab59d9795358a43a65911c0/controls/python_works.rb#L60-L67)

File: controls/gdal_works.rb [hunk 2]
```diff
@@ -19,13 +19,180 @@
...
+      its('stdout') { should match /GDAL #{plan_pkg_version}/ }
+      its('stderr') { should be_empty }
+  end
+  ogrlineref_full_path = File.join(plan_installation_directory.stdout.strip, "bin/ogrlineref")
+  describe command("#{ogrlineref_full_path} --version") do
+      its('exit_status') { should eq 0 }
+      its('stdout') { should_not be_empty }
+      its('stdout') { should match /GDAL #{plan_pkg_version}/ }
+      its('stderr') { should be_empty }
+  end
```
Comment (MindNumbing) [inline controls/gdal_works.rb]: Can you consolidate and loop over these tests? See example [here](https://github.com/chef-base-plans/python2/blob/07207f3ce5f3fe713ab59d9795358a43a65911c0/controls/python_works.rb#L69-L75)

**Review by MindNumbing — CHANGES_REQUESTED** [review]
Some changes to consider, otherwise looks great.

File: controls/gdal_exists.rb (hunk 1 above)
Comment (gavindidrichsen) [reply controls/gdal_exists.rb]: All 'exists' tests combined into a single array loop.

File: controls/gdal_works.rb (hunk 2 above)
Comment (gavindidrichsen) [reply controls/gdal_works.rb]: All 'works' tests combined into single array loop apart from a few exceptions that need slight variations on the checks

**Review by MindNumbing — APPROVED** [review]
Thanks @gavindidrichsen, LGTM! 👍

**Pull request merged**